python/tritonserver/_api/_response.py

```diff
@@ -0,0 +1,386 @@
+# Copyright 2023-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Class for receiving inference responses to Triton Inference Server Models"""
+
+from __future__ import annotations
+
+import asyncio
+import inspect
+import queue
+from typing import Optional
+
+from _model import Model
+from tritonserver._c.triton_bindings import InternalError, TRITONSERVER_InferenceRequest
+from tritonserver._c.triton_bindings import TRITONSERVER_LogLevel as LogLevel
+from tritonserver._c.triton_bindings import TRITONSERVER_LogMessage as LogMessage
+from tritonserver._c.triton_bindings import (
+    TRITONSERVER_ResponseCompleteFlag,
+    TRITONSERVER_Server,
+)
+
+
+class AsyncResponseIterator:
+
+    """Asyncio compatible response iterator
+
+    Response iterators are returned from model inference methods and
+    allow users to process inference responses in the order they were
+    received for a request.
+
+    """
+
+    def __init__(
+        self,
+        _model: Model,
```

Comment:
I think constructor/method arguments shouldn't start with underscore.

Reply:
mainly wanted to indicate these were private - but will switch back, I think the docstring + import should be enough. Mainly the object is intended to be gotten from a factory method and not directly instantiated by user.